source/base/multithread_info.cc

```diff
@@ -143,6 +146,26 @@ MultithreadInfo::initialize_multithreading()
   done = true;
 }
 
+
+
+#ifdef DEAL_II_WITH_TBB
+tbb::task_arena &
+MultithreadInfo::get_tbb_task_arena()
+{
+  // This should not trigger in normal user code, because we initialize the
+  // Executor in the static DoOnce struct at the end of this file unless you
+  // ask for the Executor before this static object gets constructed.
+  Assert(
+    task_arena.get() != nullptr,
+    ExcMessage(
+      "Please initialize multithreading using MultithreadInfo::set_thread_limit() first."));
+  return *(task_arena.get());
+}
+std::unique_ptr<tbb::task_arena> MultithreadInfo::task_arena = nullptr;
```

Comment:
I *think* that the only place where you really access the `task_arena` variable is in this function. If this is true, a maybe simpler design would be to use a `static` local variable in this function, rather than a global pointer. This way you can remove the definition here in line 164, and the declaration in the header file, and you don't have to check via the assertion in the implementation of the function because you know that you must have gotten across the initialization if you get to the point where the variable is used.

Reply:
Yes, I agree. This is a currently a verbatim copy of what we have for the `Executor` class from taskflow and will probably go away again in the final version of this PR.